app/src/main/java/org/kinecosystem/kinit/server/OnboardingService.kt

```diff
@@ -139,11 +138,10 @@ class OnboardingService(context: Context, private val appLaunchApi: OnboardingAp
                 if (response.isSuccessful) {
                     response.body()?.list?.let {
                         blackList = it
-                        if (!blackList.contains(DeviceUtils.getLocalDialPrefix(applicationContext))) {
-                            isInBlackList = false
-                            if(!userRepo.isRegistered) {
-                                callRegister(BuildConfig.VERSION_NAME)
-                            }
+                        if (blackList.contains(DeviceUtils.getLocalDialPrefix(applicationContext))) {
+                            isInBlackList = true
```

Comment:
Following is clearer and simpler: 
`isInBlackList = blackList.contains(DeviceUtils.getLocalDialPrefix(applicationContext))`
`if (!isInBlackList && !userRepo.isRegistered) {
}`